src/shims/aarch64.rs

```diff
@@ -0,0 +1,69 @@
+use std::cmp;
+
+use rustc_middle::ty::Ty;
+use rustc_span::Symbol;
+use rustc_target::callconv::{Conv, FnAbi};
+
+use self::helpers::bool_to_simd_element;
+use crate::*;
+
+impl<'tcx> EvalContextExt<'tcx> for crate::MiriInterpCx<'tcx> {}
+pub(super) trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
+    fn emulate_aarch64_intrinsic(
+        &mut self,
+        link_name: Symbol,
+        abi: &FnAbi<'tcx, Ty<'tcx>>,
+        args: &[OpTy<'tcx>],
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx, EmulateItemResult> {
+        let this = self.eval_context_mut();
+        // Prefix should have already been checked.
+        let unprefixed_name = link_name.as_str().strip_prefix("llvm.aarch64.").unwrap();
+        match unprefixed_name {
+            "isb" => {
+                let [arg] = this.check_shim(abi, Conv::C, link_name, args)?;
+                let arg = this.read_scalar(arg)?.to_i32()?;
+                match arg {
+                    // SY ("full system scope")
+                    15 => {
+                        this.yield_active_thread();
+                    }
+                    _ => {
+                        throw_unsup_format!("unsupported llvm.aarch64.isb argument {}", arg);
+                    }
+                }
+            }
+
+            "neon.umaxp.v16i8" => {
+                let [left, right] = this.check_shim(abi, Conv::C, link_name, args)?;
+
+                let (left, left_len) = this.project_to_simd(left)?;
+                let (right, right_len) = this.project_to_simd(right)?;
+                let (dest, lane_count) = this.project_to_simd(dest)?;
+                assert_eq!(left_len, right_len);
+                assert_eq!(lane_count, left_len);
+
+                for lane_idx in 0..lane_count {
+                    let src = if lane_idx < (lane_count / 2) { &left } else { &right };
+                    let src_idx = lane_idx % (lane_count / 2);
+
+                    let lhs_lane = this.read_scalar(&this.project_index(src, src_idx * 2)?)?;
+                    let rhs_lane = this.read_scalar(&this.project_index(src, src_idx * 2 + 1)?)?;
+
+                    let res_lane = match unprefixed_name {
+                        "neon.umaxp.v16i8" => {
+                            Scalar::from_i8(cmp::max(lhs_lane.to_i8()?, rhs_lane.to_i8()?))
+                        }
```

Comment:
If this ends up existing for a bunch of types, it's probably better to use a generic implementation directly on `ImmTy`, using the interpreter methods for arithmetic on immediates.

Reply:
Which function should I use for that?

Reply:
`this.binary_op`